test/transformer/transformer_maps_test.dart

```diff
@@ -19,11 +18,22 @@ part "transformer_test_core.dart";
 /// We pretend that our main rule, [ThingWithDateSerializatinoRule] is generated
 /// so we can't or don't want to just modify it, so instead we subclass and
 /// modify the data before calling it.
-class SpecialThingWithDateRule extends ThingWithDateSerializationRule {
+class SpecialThingWithDateRule extends CustomRule {
+
+  CustomRule generated;
```

Comment:
Collisions should be rare. I think the best option would be to have a parameter on the annotation. So, e.g. @Serializable(ruleName: 'FooRuleWithMyName')